lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -23,34 +23,172 @@ using namespace mlir::torch;
 using namespace mlir::torch::Torch;
 
 namespace {
-#define DECL_CONVERT_ATENOP(aten_op)                                              \
-class ConvertAten##aten_op##Op : public OpConversionPattern<Aten##aten_op##Op> {  \
-public:                                                                           \
-  using OpConversionPattern::OpConversionPattern;                                 \
-  LogicalResult                                                                   \
-  matchAndRewrite(Aten##aten_op##Op op, ArrayRef<Value> operands,                 \
-                  ConversionPatternRewriter &rewriter) const override;            \
-}; 
+// These legalizations are for unary ops with only for FP datatypes.
+// There is no supported quantized integer mode for these.
+#define DEF_FULLCONV_FPONLY_UNARY_ATENOP(aten_op, tosa_op)                     \
+  class ConvertAten##aten_op##Op                                               \
+      : public OpConversionPattern<Aten##aten_op##Op> {                        \
+  public:                                                                      \
+    using OpConversionPattern::OpConversionPattern;                            \
+    LogicalResult                                                              \
+    matchAndRewrite(Aten##aten_op##Op op, ArrayRef<Value> operands,            \
+                    ConversionPatternRewriter &rewriter) const override {      \
+      Aten##aten_op##Op::Adaptor adaptor(operands);                            \
+      Value self = adaptor.self();                                             \
+      auto selfTy = self.getType().cast<TensorType>();                         \
+      if (selfTy) {                                                            \
```

Comment:
I think we can do an early exit over here to save indentation. `if(!selfTy) op.emitError...`.